var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -408,39 +406,35 @@ def setup_build_tests(self):
         self.cache_extra_test_sources(cmake_out_path)
         self.cmake_bin(set=True)
 
-    def build_tests(self, cmake_path):
+    def test_build(self):
         """Build test."""
-        cmake_bin = self.cmake_bin(set=False)
-
-        if not cmake_bin:
-            tty.msg("Skipping kokkos test: cmake_bin_path.txt not found")
-            return
```

Comment:
This is definitely the `cmake` hack that needs to be removed -- along with the `cmake_bin` method (lines 375-387) -- but the substitution is `self.spec["cmake"].command`.  See the second note in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#building-and-running-test-executables.